src/libasr/ASR.asdl

```diff
@@ -247,6 +247,7 @@ expr
     | StringRepeat(expr left, expr right, ttype type, expr? value)
     | StringLen(expr arg, ttype type, expr? value)
     | StringItem(expr arg, expr idx, ttype type, expr? value)
+    | StringSection(expr arg, array_index section, ttype type, expr? value)
```

Comment:
I would probably do it like this:
```suggestion
    | StringSection(expr arg, expr start, expr end, expr step, ttype type, expr? value)
```
And `step` is always present, and equal to 1 if it is not specified in AST explicitly.